drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/referral_postings_loader.rb

```diff
@@ -0,0 +1,229 @@
+    end
+
+    def build_household_member_records
+      client_ids_by_mci_id = Hmis::Hud::Client
+        .where(data_source: data_source)
+        .pluck(:personal_id, :id)
+        .to_h
+      referral_pks_by_referral_id = referral_scope.pluck(:identifier, :id).to_h
+      record_class = HmisExternalApis::AcHmis::ReferralHouseholdMember
+      household_member_rows.map do |row|
+        mci_id = row_value(row, field: 'MCI_ID')
+        referral_id = row_value(row, field: 'REFERRAL_ID')
+        record_class.new(
+          referral_id: referral_pks_by_referral_id.fetch(referral_id),
+          relationship_to_hoh: relationship_to_hoh(row),
+          mci_id: mci_id,
+          client_id: client_ids_by_mci_id.fetch(mci_id),
+        )
+      end
+    end
+
+    def build_posting_records
+      record_class = HmisExternalApis::AcHmis::ReferralPosting
+      referral_pks_by_referral_id = referral_scope.pluck(:identifier, :id).to_h
+      projects_by_project_id = Hmis::Hud::Project
+        .where(data_source: data_source)
+        .pluck(:project_id, :id)
+        .to_h
+      unit_types_by_mper = Hmis::UnitType
+        .joins(:mper_id)
+        .pluck('external_ids.value', :id)
+        .to_h
+      posting_rows.map do |row|
+        referral_id = row_value(row, field: 'REFERRAL_ID')
+        record = record_class.new(
+          referral_id: referral_pks_by_referral_id.fetch(referral_id),
+          data_source_id: data_source.id,
+          identifier: row_value(row, field: 'POSTING_ID'),
+          status: posting_status(row),
+          project_id: projects_by_project_id.fetch(row_value(row, field: 'PROGRAM_ID')),
+          unit_type_id: unit_types_by_mper.fetch(row_value(row, field: 'UNIT_TYPE_ID')),
+          resource_coordinator_notes: row_value(row, field: 'RESOURCE_COORDINATOR_NOTES', required: false),
+          HouseholdID: referral_household_id(referral_id),
+        )
+        # not totally sure how to treat these dates
+        if record.assigned_status?
+          record.status_updated_at = parse_date(row_value(row, field: 'ASSIGNED_AT'))
+        else
+          record.status_updated_at = parse_date(row_value(row, field: 'STATUS_UPDATED_AT'))
+        end
+        record
+      end
+    end
+
+    def relationship_to_hoh(row)
+      @posting_status_map ||= HmisExternalApis::AcHmis::ReferralHouseholdMember
+        .relationship_to_hohs
+        .invert.stringify_keys
+      # the field name here is different from the spec
+      @posting_status_map.fetch(row_value(row, field: 'RELATIONSHIP_TO_HOH_ID'))
+    end
+
+    def posting_status(row)
+      value = row_value(row, field: 'STATUS')
+      return unless value
+
+      value = value.downcase.gsub(' ', '_') + '_status'
+      HmisExternalApis::AcHmis::ReferralPosting.statuses.fetch(value)
+    end
+
+    def referral_scope
+      enrollments = Hmis::Hud::Enrollment.where(data_source: data_source)
+      referral_class.where(enrollment_id: enrollments.select(:id))
```

Comment:
```suggestion
      referral_class.where(enrollment_id: enrollments.pluck(:id))
```
since we don't need the AR records?

Reply:
I like to use select rather than pluck since AR turns that into a subquery which is usually (but not always) more efficient.

However in this case I just removed the DS condition since the relation between DS and referral is a little awkward and there shouldn't be any other referrals